platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/PropertyFactory.java

```diff
@@ -2375,6 +2375,26 @@
     return new LayoutPropertyValue<>("text-max-angle", value);
   }
 
+  /**
+   * The property allows to control an orientation of a symbol. Note, that the property values act as a hint, so that Symbols whose language doesn't support provided orientation, will be laid out in their natural orientation. Example: English point symbol will be rendered horizontally even if `"text-writing-mode": ["vertical"]` is set. The order of elements in an array define priority order for the placement of an orientation variant.
+   *
+   * @param value a String[] value
+   * @return property wrapper around String[]
+   */
+  public static PropertyValue<String[]> textWritingMode(String[] value) {
```

Comment:
the `String[]` setup matches what we do for [text-anchor](https://docs.mapbox.com/mapbox-gl-js/style-spec/#layout-symbol-text-anchor). For that property we expose an enum: `Optional enum. One of "center", "left", "right", "top", "bottom", "top-left", "top-right", "bottom-left", "bottom-right". Defaults to "center"`.

Are we going to do the same for `textWritingMode`? If yes can we generate an enum in `Property.java`?

Reply:
@tobrun I had to tweak generator, so that enum bindings would be generated for array properties that define unique enums.